### PR TITLE
INGK-790 Run no-connection tests in a Linux machine

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

No-connection tests are run in a Linux machine.

Fixes # INGK-790

### Type of change

Please add a description and delete options that are not relevant.

- [x] Add a stage in the Jenkinsfile.

### Tests
- [x] Fix some tests to avoid using files from external servers.
- [x] Run tests.

Please describe the tests that you ran to verify your changes if it applies. 

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Build documentation locally to verify changes.
- [ ] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: tests/canopen/test_canopen_network.py [hunk 1]
```diff
@@ -115,6 +117,8 @@ def test_disconnect_from_slave(read_config):
 
 @pytest.mark.no_connection
 def test_setup_and_teardown_connection(virtual_network):
+    if os.name != "nt":
```
Comment (MartinALL91) [inline tests/canopen/test_canopen_network.py:120]: Use https://docs.python.org/3/library/platform.html#platform.system

File: virtual_drive/core.py [hunk 2]
```diff
@@ -1281,6 +1281,9 @@ def __init__(self, port: int, dictionary_path: Optional[str] = None) -> None:
         self.internal_generator = VirtualInternalGenerator(self)
 
         self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        if os.name != "nt":
```
Comment (MartinALL91) [inline virtual_drive/core.py:1284]: Check if OS is linux

Comment (pzinemanas) [reply virtual_drive/core.py]: Done [db84905](https://github.com/ingeniamc/ingenialink-python/pull/391/commits/db849054c96971aa40e6b4216c347b3bca9990aa)

File: tests/canopen/test_canopen_network.py (hunk 1 above)
Comment (pzinemanas) [reply tests/canopen/test_canopen_network.py]: Done [db84905](https://github.com/ingeniamc/ingenialink-python/pull/391/commits/db849054c96971aa40e6b4216c347b3bca9990aa)

**Review by GullinBustin — APPROVED** [review]

**Review by MartinALL91 — APPROVED** [review]

**Pull request merged**